pkg/clients/iam/tags.go

```diff
@@ -0,0 +1,64 @@
+package iam
+
+import (
+	"github.com/aws/aws-sdk-go-v2/aws"
+	iamtypes "github.com/aws/aws-sdk-go-v2/service/iam/types"
+
+	"github.com/crossplane/provider-aws/apis/iam/v1beta1"
+)
+
+// DiffIAMTags returns the lists of tags that need to be removed and added according
+// to current and desired states, also returns if desired state needs to be updated
+func DiffIAMTags(local map[string]string, remote []iamtypes.Tag) (add []iamtypes.Tag, remove []string, areTagsUpToDate bool) {
+	addMap, remove, areTagsUpToDate := diffTags(local, remote)
+
+	for k, v := range addMap {
+		add = append(add, iamtypes.Tag{Key: aws.String(k), Value: aws.String(v)})
+	}
+	return add, remove, areTagsUpToDate
+}
+
+// DiffIAMTagsWithUpdates returns the lists of tags that need to be removed and added according
+// to current and desired states;
+// tags that have changed will be returned in the addOrUpdate return parameter, but not included in the `remove` return parameters
+// it also returns if desired state needs to be updated
+func DiffIAMTagsWithUpdates(local []v1beta1.Tag, remote []iamtypes.Tag) (addOrUpdate []iamtypes.Tag, remove []string, areTagsUpToDate bool) {
```

Comment:
Shouldn't we use this for all cases? Do other API endpoints in IAM return error when you call `Tag` for an existing tag?